src/main/scala/scala/tools/refactoring/analysis/SymbolExpanders.scala

```diff
@@ -95,6 +94,107 @@ trait DependentSymbolExpanders extends TracingImpl {
     }
   }
 
+  /**
+   * Associates class vals with constructor parameters
+   */
+  trait ClassVals extends SymbolExpander { this: IndexLookup =>
+    protected abstract override def doExpand(s: Symbol) = {
+      findRelatedCtorParamSymbol(s).toList ::: super.doExpand(s)
+    }
+
+    private def findRelatedCtorParamSymbol(s: Symbol): Option[Symbol] = s match {
+      case ts: TermSymbol if ts.isVal && ts.owner.isClass =>
+        declaration(s.owner).flatMap(findRelatedCtorParamSymbolIn(_, s))
+      case _ => None
+    }
+
+    private def findRelatedCtorParamSymbolIn(parent: Tree, valSym: Symbol): Option[Symbol] = {
+      parent.foreach {
+        case dd: DefDef if dd.symbol.isConstructor =>
+          val res = findParamSymbolAssociatedWithValSymbol(dd, valSym)
+
+          if (res.nonEmpty) {
+            return res
```

Comment:
`parent.iterator.collect{ case dd: DefDef if dd.symbol.isConstructor => findParamSymbolAssociatedWithValSymbol(dd, valSym)}.find(_.nonEmpty)` should do the same